python/beeai_framework/adapters/litellm/chat.py

```diff
@@ -289,5 +287,49 @@ async def clone(self) -> Self:
         cloned.model_supports_tool_calling = self.model_supports_tool_calling
         return cloned
 
+    def _assert_setting_value(
+        self,
+        name: str,
+        value: Any | None = None,
+        *,
+        constructor_name: str | None = None,
+        aliases: list[str] | None = None,
+        envs: list[str],
+        fallback: str | None = None,
+        allow_none: bool = False,
+    ) -> None:
+        def reset_aliases() -> None:
+            for alias in aliases or []:
+                self._settings[alias] = None
+
+        if value is not None:
+            self._settings[name] = value
+
+        if self._settings.get(name):
+            reset_aliases()
+            return None
+
+        for env in envs:
+            value = os.environ.get(env)
+            if value:
+                reset_aliases()
+                self._settings[name] = value
+                return None
+
+        if fallback:
+            reset_aliases()
+            self._settings[name] = fallback
+            return None
+
+        if allow_none:
+            reset_aliases()
+            return None
+
+        raise ValueError(
+            f"Setting {constructor_name or name} is required for {type(self).__name__}. "
```

Comment:
is `constructor_name` just used here in the error message? I'm a bit confused on it's purpose and difference from `name`